app/ts/utils/bigint.ts

```diff
@@ -176,3 +176,32 @@ export function isHexEncodedNumber(input: string): boolean {
 	const hexNumberRegex = /^(0x)?[0-9a-fA-F]+$/
 	return hexNumberRegex.test(input)
 }
+
+export function calculateWeightedPercentile(data: bigint[], weights: bigint[], percentile: bigint): bigint {
```

Comment:
As mentioned above, this feels like it should take in a single array of type `readonly [bigint, bigint][]` or `readonly {effectivePriorityFee, gasUsed}[]` (`readonly` is less important, but good practice in general).